### ISSUE TITLE
Rule `E203` conflicts with formatter

### ISSUE THREAD

**Issue by ofek** [astral-sh/ruff#8642]
Running the formatter on

```python
release_lines = history_file_lines[history_file_lines.index('## Unreleased') + 1: -1]
```

produces

```python
release_lines = history_file_lines[history_file_lines.index('## Unreleased') + 1 : -1]
```

which then fails the rule

```
scripts\utils.py:22:85: E203 Whitespace before ':'
```

Is the fix simply to document that here? https://docs.astral.sh/ruff/formatter/#conflicting-lint-rules

**Comment by charliermarsh** [comment]
I will add it to that list, though there's an open issue to fix the rule to allow this: https://github.com/astral-sh/ruff/issues/7259.

**Comment by charliermarsh** [comment]
I decided to just fix it instead.

**Comment by ofek** [comment]
Oh nice that's much better, thanks!

**Comment by JohannesBuchner** [comment]
I am still getting this unwanted behaviour with ruff-0.9.3:

```
$ cat > foo.py
lanes[laneid + 1:]
$ ruff format foo.py
1 file reformatted
$ cat foo.py
lanes[laneid + 1 :]
```

**Comment by dhruvmanila** [comment]
@JohannesBuchner That's an expected behavior. What the original issue was pointing out is that `E203` rule is _removing_ the whitespace before the `:` operator while the formatter adds it - causing the conflicting behavior between the linter and the formatter.

**Comment by JohannesBuchner** [comment]
@dhruvmanila Thanks for the quick response. flake8 gives the error "E203 whitespace before ':'" with the reformatted output. I see from https://github.com/psf/black/issues/315 that they claim that this is an issue of pep8. There is help at https://ichard26-testblackdocs.readthedocs.io/en/refactor_docs/compatible_configs.html for configuring flake8. Thanks.
